src/Microsoft.OpenApi/CallerArgumentExpressionAttribute.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Diagnostics;
```

Comment:
should the whole file be wrapped in conditional compilation directives? (#if !NETCOREAPP3_1_OR_GREATER)
[reference](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callerargumentexpressionattribute?view=net-7.0)

Reply:
given u target netstandard2, it is not really valuable.

If it was my project i would leverage a polyfill. eg

 * https://github.com/SimonCropp/Polyfill
 * https://github.com/Sergio0694/PolySharp
 * https://github.com/Tyrrrz/PolyShim

Then u get CallerArgumentExpressionAttribute for free as well as a significant number of other newer apis+c# features

Reply:
My reasoning being: if we start enabling multi-targeting to benefit from newer features, this won't conflict with what the framework provides if conditional compilation directives are added.

Reply:
but if u use a polyfill you dont need any compilation directives. or at least significantly fewer compilation directives